tools/certreq/bin/certreq.sh

```diff
@@ -0,0 +1,198 @@
+#!/bin/bash
+
+# redirect stderr to stdout
+exec 2>&1
+
+# -e            Will make the whole script exit if a command fails.
+#               Append '|| true' if you have commands that can fail without being an issue.
+# -o pipefail   Causes a pipeline to produce a failure return code if any command errors.
+#set -eo pipefail
+
+# DEBUG
+if [ -n "${DEBUG+x}" ] ; then
+  set -x
+  #set -v
+fi
+
+# default values (can be changed by parameters)
+HOST="127.0.0.1"
+PORT="8000"
+USERNAME="admin"
+PASSWD="admin"
+
+readonly KEY_PAIR_FILE='private.key'
+readonly CSR_FILE='request.csr'
+readonly CA_FILE='ca.crt'
+readonly CERT_FILE='cert.pem'
+readonly TOKEN_FILE='token.jwt'
+
+readonly RED='\e[91m'
+readonly CYAN='\e[96m'
+readonly NC='\e[39m' # Default foreground color
+readonly BOLD='\e[1m'
+readonly NRML='\e[0m'
+
+function main() {
+
+  echo '============================================================='
+  echo 'This is a tool to request dojot certificates for IoT devices.'
+  echo '============================================================='
+
+  precondition
+
+  parseArgs "$@"
+
+  if [ -z "${JWT}" ]; then
+    getToken
+  fi
+
+  if [ -z "${DIVICE_IDS}" ]; then
```

Comment:
DEVICE_IDS
It's in multiple places, beware when changing.

Reply:
This variable is the same in all functions, I think what is missing is to define an _initial value_ for it, since it can be exported in the shell before the script is executed and thereby "hinder" the expected processing...

Reply:
That's true!